lib/ansible/modules/packaging/os/zypper.py

```diff
@@ -107,6 +108,18 @@
         required: false
         default: "no"
         choices: [ "yes", "no" ]
+    retries:
+        version_added: "2.3"
+        description:
+            - Number of retries if another instance of zypper is running. Defaults to no retries.
+        required: false
+        default: 0
+    retry_interval:
+        version_added: "2.3"
+        description:
+            - Time in seconds to wait between retries.
+        required: false
+        default: 10
```

Comment:
I didn't think of `until` to be honest and I would prefer to use existing features e.g. `until`and instead of implementing a similar feature, unless there is a pretty good reason, e.g. a default retry count.

I thought about to just have a minimal set of retries (without adding a new argument) like we see in other modules https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/apt.py#L874 if there is no downside.